dotnet/src/InternalUtilities/planning/Extensions/IKernelPluginExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+
+#pragma warning disable IDE0130
+
+// ReSharper disable once CheckNamespace - Using the main namespace
+namespace Microsoft.SemanticKernel;
+
+/// <summary>Provides extension methods for working with <see cref="IKernelPlugin"/>s and collections of them.</summary>
+internal static class IKernelPluginExtensions
+{
+    /// <summary>Gets a collection of <see cref="KernelPluginFunctionMetadata"/> instances, one for every function in every plugin in the plugins collection.</summary>
+    /// <param name="plugins">The plugins collection.</param>
+    /// <returns>A list of metadata over every function in the plugins collection</returns>
+    public static IList<KernelPluginFunctionMetadata> GetFunctionsMetadata(this IEnumerable<IKernelPlugin> plugins)
```

Comment:
I think we should also have a `public static KernelPluginFunctionMetadata GetFunctionMetadata(this IReadOnlyKernelPluginCollection plugins, string pluginName, string functionName)` that gets the KernelPluginFunctionMetadata for a specific function, with the plugin name appropriately configured.

We should also need an overload of ToOpenAIFunction that takes a KernelPluginFunctionMetadata.  That would then be used in places like:
https://github.com/microsoft/semantic-kernel/pull/3748/files#diff-e1c9cdaa9f8f5449a7cac7aab6c7f6abb673b04e3bea5de3ea16635423053f6dR170
such that the plugin name would be appropriately propagated.